vendor/github.com/openshift/library-go/pkg/manifest/manifest.go

```diff
@@ -132,15 +169,13 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
 		}
 	}
 
-	if includeTechPreview != nil {
-		featureGateAnnotation := "release.openshift.io/feature-gate"
-		featureGateAnnotationValue, featureGateAnnotationExists := annotations[featureGateAnnotation]
-		if featureGateAnnotationValue == string(configv1.TechPreviewNoUpgrade) && !(*includeTechPreview) {
-			return fmt.Errorf("tech-preview excluded, and %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	if requiredFeatureSet != nil {
+		manifestFeatureSets, manifestSpecifiesFeatureSets, err := getFeatureSets(annotations)
+		if err != nil {
+			return err
 		}
-		// never include the manifest if the feature-gate annotation is outside of allowed values (only TechPreviewNoUpgrade is currently allowed)
-		if featureGateAnnotationExists && featureGateAnnotationValue != string(configv1.TechPreviewNoUpgrade) {
-			return fmt.Errorf("unrecognized value %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+		if manifestSpecifiesFeatureSets && !manifestFeatureSets.Has(*requiredFeatureSet) {
+			return fmt.Errorf("%q is required, and %s=%s", *requiredFeatureSet, featureSetAnnotation, strings.Join(manifestFeatureSets.List(), ","))
```

Comment:
nit: this assumes `featureSetAnnotation`, and not the old annotation name.  Is the idea that we drop the old name before we GA 4.12, once existing folks have migrated?

Reply:
> nit: this assumes featureSetAnnotation, and not the old annotation name. Is the idea that we drop the old name before we GA 4.12, once existing folks have migrated?

correct